lib/rules/no-useless-escape.js

```diff
@@ -103,15 +110,31 @@ module.exports = {
          */
         function report(node, startOffset, character) {
             const start = sourceCode.getLocFromIndex(sourceCode.getIndexFromLoc(node.loc.start) + startOffset);
+            const rangeStart = sourceCode.getIndexFromLoc(node.loc.start) + startOffset;
+            const range = [rangeStart, rangeStart + 1];
 
             context.report({
                 node,
                 loc: {
                     start,
                     end: { line: start.line, column: start.column + 1 }
                 },
-                message: "Unnecessary escape character: \\{{character}}.",
-                data: { character }
+                messageId: "unnecessaryEscape",
+                data: { character },
+                suggest: [
+                    {
+                        messageId: "removeEscape",
+                        fix(fixer) {
+                            return fixer.removeRange(range);
+                        }
+                    },
+                    {
+                        messageId: "escapeBackslash",
```

Comment:
One thought going forward - do we want to create a convention that indicates which messages are for suggestions?